web/src/reducers/activities.selectors.js

```diff
@@ -0,0 +1,76 @@
+import { createSelector } from 'reselect';
+import { selectApdData } from './apd.selectors';
+
+export const selectActivityKeys = ({ activities: { allKeys } }) => allKeys;
+
+export const selectActivitiesByKey = ({ activities: { byKey } }) => byKey;
+
+export const selectActivityByKey = ({ activities: { byKey } }, { aKey }) =>
+  byKey[aKey];
+
+const selectBudgetForActivity = ({ budget }, { aKey }) =>
+  budget.activities[aKey];
+
+export const makeSelectCostAllocateFFP = () =>
+  createSelector(
+    [selectActivityByKey, selectBudgetForActivity],
+    (activity, budget) => {
+      const { costAllocation } = activity;
+      const { costsByFFY } = budget;
+
+      const byYearData = Object.entries(costsByFFY)
+        .filter(([year]) => year !== 'total')
+        .map(([year, costs]) => {
+          const { ffp } = costAllocation[year];
+          const ffpSelectVal = `${ffp.federal}-${ffp.state}`;
+
+          return {
+            year,
+            total: costs.total,
+            medicaidShare: costs.medicaidShare,
+            ffpSelectVal,
+            allocations: { federal: costs.federal, state: costs.state }
+          };
+        });
+
+      return { byYearData, costAllocation };
```

Comment:
The actual body of this selector came directly from a `mapStateToProps` function elsewhere.  Yay for not having to figure all this out again.  Also, the `mapStateToProps` tests on those components incidentally tests these selectors, so 👏 for that too.